### PR TITLE
Propagate changes to dropdowns

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

This PR does two things:
- Adds an error handler if writing to "apply changes" fails. As an example of this, use `caput %MYPVPREFIX%DAE:DAESETTINGS:SP.ASG MANAGER` to make the apply changes button "manager only", and then confirm you get a sensible error message when pressing apply changes.
- Does a bit of refactoring to make sure the tables in the dropdown menus are databound to the values in the backend.

The original issue in the ticket was caused by the values in the backend becoming out of sync with the values on the screen. The new databinding should fix that issue.

You can reproduce the original issue using two clients and two wiring tables:
- Open both clients
- In client 1, select table 1 and then press apply changes
- In client 2, select table 2 and then press apply changes
- Without changing the selection in the dropdown menu, press apply changes again in client 1.
- Bug: Table 1 is not set.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/2948

### Acceptance criteria

- Items in description above work

### Unit tests

No unit tests added.

This change is not easy to unit test as it's all in the data binding/observable layers as opposed to doing any logic.

### System tests

Squish not yet ready. RCPTT dead (and doing multi-client tests in RCPTT is very hard).

### Documentation

None

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/dataacquisition/ObservingDaeSettings.java [hunk 1]
```diff
@@ -43,7 +45,7 @@ protected String asText() {
 		};
 	}
 	
-	public void sendUpdate() {
+	public void sendUpdate() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/dataacquisition/ObservingDaeSettings.java]: Could you add Javadoc please?

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/periods/ObservingPeriodSettings.java [hunk 2]
```diff
@@ -43,7 +45,7 @@ protected String asText() {
 		};
 	}
 	
-	public void sendUpdate() {
+	public void sendUpdate() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/periods/ObservingPeriodSettings.java]: Javadoc

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/timechannels/ObservingTimeChannels.java [hunk 3]
```diff
@@ -42,7 +44,7 @@ protected String asText() {
 		};
 	}
 	
-	public void sendUpdate() {
+	public void sendUpdate() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/timechannels/ObservingTimeChannels.java]: Javadoc

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/ExperimentSetup.java [hunk 4]
```diff
@@ -175,11 +177,13 @@ public PeriodSettings periodSettings() {
     /**
      * Sends all changes made to the experiment setup in the GUI to the server.
      */
-	public void sendAllSettings() {
+	public synchronized void sendAllSettings() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/experimentsetup/ExperimentSetup.java]: Checkstyle complaining about missing `@throws` tag in doc

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/internal/SettingsGateway.java [hunk 5]
```diff
@@ -58,8 +58,8 @@ public SettingsGateway(ForwardingObservable<String> settingsSource, Writable<Str
 	/**
 	 * Sends the update.
 	 */
-	public void sendUpdate() {
-		updateWriter.uncheckedWrite(asText());
+	public synchronized void sendUpdate() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/internal/SettingsGateway.java]: Checkstyle complaining about missing @throws tag in doc

File: base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/updatesettings/ObservingUpdateSettings.java [hunk 6]
```diff
@@ -19,10 +19,13 @@
 
 package uk.ac.stfc.isis.ibex.dae.updatesettings;
 
+import java.io.IOException;
+
 import uk.ac.stfc.isis.ibex.dae.internal.SettingsGateway;
 import uk.ac.stfc.isis.ibex.epics.observing.ForwardingObservable;
 import uk.ac.stfc.isis.ibex.epics.pv.Closable;
 import uk.ac.stfc.isis.ibex.epics.writing.Writable;
+import uk.ac.stfc.isis.ibex.logger.IsisLog;
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/updatesettings/ObservingUpdateSettings.java]: unused import

File: base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/ExperimentSetup.java [hunk 7]
```diff
@@ -19,9 +19,12 @@
 
 package uk.ac.stfc.isis.ibex.ui.dae.experimentsetup;
 
+import java.io.IOException;
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/ExperimentSetup.java]: Unused import

File: base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/ExperimentSetupViewModel.java [hunk 8]
```diff
@@ -61,7 +63,7 @@ public void setModel(ExperimentSetup model) {
      * updates the experiment setup saved in the blockserver with changes made
      * in the frontend.
      */
-	public void updateDae() {
+	public void updateDae() throws IOException {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/experimentsetup/ExperimentSetupViewModel.java]: Checkstyle complaining about missing @throws tag in doc

**Comment by ThomasLohnert** [comment]
Some of the tests are failing due to the blank line being taken out, could you please fix them? Rest is all good.

**Comment by Tom-Willemsen** [comment]
Tests fixed

**Pull request merged**